frontend/src/pages/home/MainPage.tsx

```diff
@@ -1,19 +1,15 @@
 //This main page is used to test on implementing main page including dashboard, team, project, etc
 import React from 'react';
-import { Sidebar } from './Sidebar';
-import Dashboard from './Dashboard';
 import { useRequireAuthenticated} from '../../hooks/auth';
+import SidebarPageTemplate from '../../templates/SidebarPageTemplate';
 
 const Main: React.FC = () => {
   // add hook requiring authentication from the user.
   const displayContent = useRequireAuthenticated()
   return (
-    displayContent &&
-    <main className='grid gap-4 p-4 grid-cols-[200px,_1fr] bg-stone-100 text-stone-950 rounded-lg'>
-      <Sidebar />
-      <Dashboard />
-    </main>
-
+    <SidebarPageTemplate hidden={!displayContent}>
```

Comment:
Just wondering what is SidebarPageTemplate?

Reply:
It's a page template that basically takes care of creating the sidebar component on the page. Since many pages need to have the sidebar in there, I created this "page template" so that other pages can reuse it